scripts/buildTokens.js

```diff
@@ -0,0 +1,118 @@
+const glob = require('glob')
+const StyleDictionary = require('style-dictionary')
+const {
+  transformScssTokens,
+  transformTailwindTokens
+} = require('./utils/transformTokens')
+
+const prefix = 'dp-'
+
+const tokensPath = 'tokens/src/**/*.json'
+
+const files = glob
+  .sync(tokensPath)
+  .map(filePath => filePath
+    .replace('tokens/src/', '')
+    .replace('color/', '')
+    .replace('.json', ''))
+    .filter(filePath => !filePath.startsWith('_')) // Do not render tokens only used internally
+
+// Define custom output groups for Tailwind, rather than just grouping them by file path
+const corePlugins = ['fontSize', 'boxShadow', 'breakpoints', 'rounded', 'space', 'zIndex']
+const corePluginsColor = ['color', 'textColor', 'backgroundColor', 'borderColor']
+
+// Adds prefix to variables. Keeps uppercase path parts, which is needed later.
+StyleDictionary.registerTransform({
+  name: 'name/scss',
+  type: 'name',
+  transformer: (token) => prefix + token.path.join('-')
+})
+
+// Adds prefix transform to default scss transform group
+StyleDictionary.registerTransformGroup({
+  name: 'custom/scss',
+  transforms: StyleDictionary.transformGroup.scss.concat(['name/scss'])
+})
+
+// Adds prefix to variables. Keeps uppercase path parts, which is needed later.
+StyleDictionary.registerTransform({
+  name: 'name/web',
+  type: 'name',
+  transformer: (token) => token.path.join('-')
+})
+
+// Adds prefix transform to default scss transform group
+StyleDictionary.registerTransformGroup({
+  name: 'custom/web',
+  transforms: StyleDictionary.transformGroup.web.concat(['name/web'])
+})
+
+/**
+ * Custom format that generates javascript modules ready to be used as Tailwind config files.
+ * The values use a nested set of css variables to enable theming on multiple levels.
+ */
+StyleDictionary.registerFormat({
+  name: 'tailwind/variables',
+  formatter: (formatterArguments) => transformTailwindTokens(formatterArguments, corePluginsColor, corePlugins)
+})
+
+/**
+ * This format is a customized version of the implementation in `formattedVariables`.
+ * It applies some transformations to scss variables according to the conventions in demosplan-ui.
+ * See https://github.com/amzn/style-dictionary/blob/17f4cb2f30bd002dfd55d6ef8c5bee4138de8d64/lib/common/formatHelpers/formattedVariables.js#L46
+ */
+StyleDictionary.registerFormat({
+  name: 'scss/customVariables',
+  formatter: transformScssTokens
+})
+
+const StyleDictionaryExtended = StyleDictionary.extend({
+  source: [tokensPath],
+  platforms: {
+    js: {
+      transformGroup: 'js',
+      buildPath: 'tokens/dist/js/',
+      files: files.map((filePath) => {
+        return {
+          destination: `${filePath}.js`,
+          format: 'javascript/module',
+          filter: (token) => {
+            return token.filePath.includes(filePath) && token.$status !== 'Deprecated'
```

Comment:
I thought deprecated means its not recommended for usage and will be removed "soon", but still available. With this. Its not available at all if I am correct.

Reply:
This is correct, however the `'javascript/module'` format used in the `js` transformGroup produces js modules that are only used for the Storybook documentation.

For the Tailwind transform, i also have chosen to only output non-deprecated tokens (see `isDeprecated` in transformTokens.js). The "legacy tokens" are only available in Scss, as we're not gonna use them in Tailwind. At the moment, the `Deprecated` status is used exclusively for tokens which are renamed to match the Tailwind naming scheme, eg. in boxShadow.json, where the Scss naming scheme was `box-shadow-1`..`box-shadow-4` while Tailwind uses T-Shirt-Sizes. The deprecated tokens are simply aliases to the new names.